### PR TITLE
test: use jest instead of mocha for the test base

### PR DESCRIPTION
**Pull request by arnaudbesnier**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -2,13 +2,27 @@ module.exports = {
...
+        devDependencies: [
+          'scripts/*.js',
+          'test/**/*.js'
+        ]
+      },
+    ],
+    'jest/no-hooks': ['error', {
+      allow: ['beforeEach'],
+    }],
+    'jest/no-standalone-expect': 0,
```
Comment (arnaudbesnier) [inline .eslintrc.js]: Needs to be deactivated to work with oclif tests.

File: .eslintrc.js [hunk 2]
```diff
@@ -2,13 +2,27 @@ module.exports = {
...
+      'error',
+      {
+        devDependencies: [
+          'scripts/*.js',
+          'test/**/*.js'
+        ]
+      },
+    ],
+    'jest/no-hooks': ['error', {
+      allow: ['beforeEach'],
```
Comment (arnaudbesnier) [inline .eslintrc.js]: Needs to be deactivated to prevent authentication context issues in some test files.

File: .eslintrc.js [hunk 3]
```diff
@@ -2,13 +2,27 @@ module.exports = {
...
     'implicit-arrow-linebreak': 0,
+    'import/no-extraneous-dependencies': [
+      'error',
+      {
+        devDependencies: [
+          'scripts/*.js',
+          'test/**/*.js'
+        ]
+      },
+    ],
```
Comment (arnaudbesnier) [inline .eslintrc.js]: Add a missing basic rule to enable required dev-dependencies in scripts and tests directories

**Comment by slimee** [comment]
I think we should merge this state and continue to migrate tests from time to time.
 Wdyt?

**Comment by slimee** [comment]
> We should merge this.
> The state of this PR is:
> -rebased,
> -build and pass test and lint (at least as configured),
> -Contain many test migrations as they should be written now.
> 
> We can then:
> -write new test style,
> -migrate old ones from time to time. (very quick now 👍 )
> -unrelove lint rules from time to time.
> 
> 🚀 ? @arnaudbesnier

It must be rebased again.

**Comment by slimee** [comment]
It's rebased, there is now only 3 test files that still use the old 'oclif/test' way of testing.

**Comment by slimee** [comment]
get rid of @oclif/test, which was not living.
All tests use now testCli method.

**Pull request merged**